### PR TITLE
Updated the new endpoint type AzureArcNetwork and some additional properties

### PR DESCRIPTION
**Pull request by v-sapanchal**
…uired properties

# ARM (Control Plane) API Specification Update Pull Request 

> Tip: overwhelmed by all this guidance? See the `Getting help` section at the bottom of this PR description.

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

<details>
<summary> Click here to see the details of Step 1 </summary>

### Breaking changes review (Diagram Step 1)

If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
- The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
- You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

</details>

<details> 
<summary> Click here to see the details of Step 2 </summary>

### ARM API changes review (Diagram Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

</details>

<details>
<summary> Click here to see the diagram footnotes</summary>

### Diagram footnotes

[1] [ARM review queue] (for **merge** queues, see [2])  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.
[2] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])

If you need further help with anything, see `Getting help` section below.

</details>

## Purpose of this PR

What's the purpose of this PR? Check all that apply. This is **mandatory**!

  - [ ] New API version. (If API spec is not defined in TypeSpec, the PR should have been generated using [OpenAPI Hub](https://aka.ms/openapihub)).
  - [x] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [x] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to the Diagram Step 2, "ARM API changes Review", for this PR.

## Additional information

<details>
<summary> Viewing API changes</summary>

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

</details>
<details>
<summary>Suppressing failures</summary>

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/azsdk/pr-suppressions) 
to get approval.

</details>

## Getting help

- First, please carefully read through this PR description, from top to bottom. Please fill out the `Purpose of this PR` and `Due diligence checklist`.
- To understand what you must do next to merge this PR, see the `Next Steps to Merge` comment. It will appear within few minutes of submitting this PR and will continue to be up-to-date with current PR state.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the PR CI checks appear to be stuck in `queued` state, please add a comment with contents `/azp run`.
  This should result in a new comment denoting a `PR validation pipeline` has started and the checks should be updated after few minutes.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Review by kumaam — APPROVED** [review]

File: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json [hunk 1]
```diff
@@ -4273,6 +4274,14 @@
...
           "description": "Test coverage for the endpoint."
+        },
+        "locationDetails": {
+          "$ref": "#/definitions/ConnectionMonitorEndpointLocationDetails",
+          "description": "Location details for connection monitor endpoint."
+        },
+        "subscriptionId": {
+          "type": "string",
+          "description": "Subscription ID for connection monitor endpoint."
         }
```
Comment (TimLovellSmith) [inline specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4285]: its not the subscription ID of the connection monitor but the monitored endpoint? Which may be in some other subscription? 

Why is this needed can it be the same as resourceID?

Comment (TimLovellSmith) [reply specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4285]: Is this only for the ARC scenario? A: YES

Comment (TimLovellSmith) [reply specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4285]: You mentioned public docs already doc this, I don't see how that is possible using proper process, please share a link.

**Comment by TimLovellSmith** [comment]
> 

can you please reflect these changes in your examples?

---
Refers to: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4802 in 1f2768e. [](commit_id = 1f2768e3864c9dec5917626d3c8c78e58a2175a5, deletion_comment = False)

File: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json [hunk 2]
```diff
@@ -4273,6 +4274,14 @@
             "modelAsString": true
           },
           "description": "Test coverage for the endpoint."
+        },
+        "locationDetails": {
+          "$ref": "#/definitions/ConnectionMonitorEndpointLocationDetails",
```
Comment (TimLovellSmith) [inline specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4279]: >ConnectionMonitorEndpointLocationDetails [](http://example.com/codeflow?start=33&length=40)

why will location details need more than just a region string?

File: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json [hunk 3]
```diff
@@ -4777,6 +4786,16 @@
         }
       },
       "description": "List of connection states snapshots."
+    },
+    "ConnectionMonitorEndpointLocationDetails": {
+      "type": "object",
+      "properties": {
+        "region": {
```
Comment (TimLovellSmith) [inline specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4793]: >region [](http://example.com/codeflow?start=9&length=6)

Is this an azure location? Why not call it 'location'?

File: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json [hunk 4]
```diff
@@ -4273,6 +4274,14 @@
...
           },
           "description": "Test coverage for the endpoint."
+        },
+        "locationDetails": {
+          "$ref": "#/definitions/ConnectionMonitorEndpointLocationDetails",
+          "description": "Location details for connection monitor endpoint."
+        },
+        "subscriptionId": {
+          "type": "string",
+          "description": "Subscription ID for connection monitor endpoint."
```
Comment (TimLovellSmith) [inline specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4284]: >"Subscription ID for connection monitor endpoint." [](http://example.com/codeflow?start=25&length=50)

In description please explain it better when it is required or optional, and what exactly it corresponds to

**Comment by TimLovellSmith** [comment]
>         "address": {

explain which types its needed or valid for? If not all

---
Refers to: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4250 in 1f2768e. [](commit_id = 1f2768e3864c9dec5917626d3c8c78e58a2175a5, deletion_comment = False)

**Comment by TimLovellSmith** [comment]
>           "description": "Filter for sub-items within the endpoint."

could reword this to mean something more like 'which endpoints within one target resource to monitor, when the target resource has multiple endpoints'

---
Refers to: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4256 in 1f2768e. [](commit_id = 1f2768e3864c9dec5917626d3c8c78e58a2175a5, deletion_comment = False)

File: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json (hunk 1 above)
Comment (v-sapanchal) [reply specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4285]: > its not the subscription ID of the connection monitor but the monitored endpoint? Which may be in some other subscription?
> 
> Why is this needed can it be the same as resourceID?

As you can see, its related to specific endpoint type of connection monitor ,
for "AzureArcNetwork" like other endpoints (-AzureVM/-ExternalAddress/-AzureArcVM and others).
for info [link](https://learn.microsoft.com/en-us/powershell/module/az.network/new-aznetworkwatcherconnectionmonitorendpointobject?view=azps-11.1.0)
For this endpoint type "AzureArcNetwork", we need some validation that requires this subscriptionId.

Comment (v-sapanchal) [reply specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4285]: > Is this only for the ARC scenario? A: YES

Yes, for "AzureArcNetwork" type.

Comment (v-sapanchal) [reply specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4285]: > You mentioned public docs already doc this, I don't see how that is possible using proper process, please share a link.

Not able to understand this query, I am not sure if you are looking for [this](https://learn.microsoft.com/en-us/powershell/module/az.network/new-aznetworkwatcherconnectionmonitorendpointobject?view=azps-11.1.0).

**Comment by v-sapanchal** [comment]
> > 
> 
> can you please reflect these changes in your examples?
> 
> Refers to: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4802 in [1f2768e](https://github.com/Azure/azure-rest-api-specs/commit/1f2768e3864c9dec5917626d3c8c78e58a2175a5). [](commit_id = [1f2768e](https://github.com/Azure/azure-rest-api-specs/commit/1f2768e3864c9dec5917626d3c8c78e58a2175a5), deletion_comment = False)

Here is the example of this change while creating the endpoint object,

$SrcEndpointScopeItem1 = New-AzNetworkWatcherConnectionMonitorEndpointScopeItemObject -Address "10.97.0.0/16"
$SourceEndpointObject1 = **_New-AzNetworkWatcherConnectionMonitorEndpointObject_** -Name "ArcNetworkAsSource" -**_AzureArcNetwork_** -IncludeItem $SrcEndpointScopeItem1 -**Location** "centraluseuap" -**SubscriptionId** "9cece3e3-0f7d-47ca-af0e-9772773f90b7"

File: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json (hunk 2 above)
Comment (kumaam) [reply specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4279]: LocationDetails can also store availabilityZone. 

Connection Monitor provides unified, end-to-end connection monitoring in Azure Network Watcher. It monitors connectivity between two endpoints,. Endpoints can be of compound type like multiple machines in one endpoint. Incase of compound resource AzureArcNetwork, this location details would be needed

File: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json (hunk 1 above)
Comment (kumaam) [reply specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4285]: subscriptionId is not of the connection monitor's subscriptionId.
 
To monitors onpremise machines, customer has to setup arc machines. In order to monitor arc machines from different subscription, this field is being used.
 
Changes are already made in network resource provider using which connection-monitors can be created with ArcNetwork as one endpoint,  but that is not publicly documented currently. It's in progress right now.
https://learn.microsoft.com/en-us/rest/api/network-watcher/connection-monitors/create-or-update?view=rest-network-watcher-2023-06-01&tabs=HTTP#endpointtype

File: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json (hunk 4 above)
Comment (kumaam) [reply specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4284]: Hey, yes it's optional, only needed in case of ArcNetwork.

File: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json (hunk 3 above)
Comment (v-sapanchal) [reply specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4793]: ConnectionMonitorEndpointLocationDetails can also store availabilityZone along with region.

**Comment by v-sapanchal** [comment]
> > 
> 
> can you please reflect these changes in your examples?
> 
> Refers to: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4802 in [1f2768e](https://github.com/Azure/azure-rest-api-specs/commit/1f2768e3864c9dec5917626d3c8c78e58a2175a5). [](commit_id = [1f2768e](https://github.com/Azure/azure-rest-api-specs/commit/1f2768e3864c9dec5917626d3c8c78e58a2175a5), deletion_comment = False)

Added a example file for ArcNetwork endpoint, "specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/examples/NetworkWatcherConnectionMonitorCreateWithArcNetwork.json"

File: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json [hunk 5]
```diff
@@ -4273,6 +4277,14 @@
             "modelAsString": true
           },
           "description": "Test coverage for the endpoint."
+        },
+        "locationDetails": {
+          "$ref": "#/definitions/ConnectionMonitorEndpointLocationDetails",
+          "description": "Location details is optional and only being used for 'AzureArcNetwork' type endpoints, which can contains availability zone along with region details."
```
Comment (piyushgupta12) [inline specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4283]: let's not put avaialbility thing for now. We only put region, for managed monitoring, we would also use Region.

File: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json [hunk 6]
```diff
@@ -4777,6 +4789,16 @@
...
+    },
+    "ConnectionMonitorEndpointLocationDetails": {
+      "type": "object",
+      "properties": {
+        "region": {
+          "type": "string",
+          "description": "Region for connection monitor endpoint."
+        }
+      },
+      "description": "Connection monitor endpoint location details only being used for 'AzureArcNetwork' type endpoints, which can contains availability zone along with region details."
```
Comment (piyushgupta12) [inline specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4801]: same here

File: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json (hunk 3 above)
Comment (raosuhas) [reply specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4796]: how would that look ? 

              "locationDetails": {
                "region": "?"
              },

Comment (v-sapanchal) [reply specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4796]: Hi,
We need other details like 'availabilityZone' along with region name, that's the reason, we are using a ConnectionMonitorEndpointLocationDetails defination for locationDetails

File: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json (hunk 6 above)
Comment (v-sapanchal) [reply specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json]: updated the description.

File: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/examples/NetworkWatcherConnectionMonitorCreateWithArcNetwork.json [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+    "subscriptionId": "subid",
+    "resourceGroupName": "rg1",
+    "networkWatcherName": "nw1",
+    "connectionMonitorName": "cm1",
+    "location": "centraluseuap",
+    "parameters": {
+      "properties": {
+        "endpoints": [
+          {
+            "name": "vm1",
```
[the remark this answers is not in the document]
Comment (v-sapanchal) [reply specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/examples/NetworkWatcherConnectionMonitorCreateWithArcNetwork.json:13]: added endpoint type.

File: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json (hunk 5 above)
Comment (v-sapanchal) [reply specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json]: updated the description.

**Comment by v-sapanchal** [comment]
> > ```
> >       "description": "Filter for sub-items within the endpoint."
> > ```
> 
> could reword this to mean something more like 'which endpoints within one target resource to monitor, when the target resource has multiple endpoints'
> 
> Refers to: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4256 in [1f2768e](https://github.com/Azure/azure-rest-api-specs/commit/1f2768e3864c9dec5917626d3c8c78e58a2175a5). [](commit_id = [1f2768e](https://github.com/Azure/azure-rest-api-specs/commit/1f2768e3864c9dec5917626d3c8c78e58a2175a5), deletion_comment = False)

Updated the description.

**Comment by v-sapanchal** [comment]
> > ```
> >       "description": "Resource ID of the connection monitor endpoint."
> > ```
> 
> explain which types its needed for? If not all
> 
> Refers to: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4248 in [1f2768e](https://github.com/Azure/azure-rest-api-specs/commit/1f2768e3864c9dec5917626d3c8c78e58a2175a5). [](commit_id = [1f2768e](https://github.com/Azure/azure-rest-api-specs/commit/1f2768e3864c9dec5917626d3c8c78e58a2175a5), deletion_comment = False)

Updated the type details in description.

**Comment by v-sapanchal** [comment]
> > ```
> >     "address": {
> > ```
> 
> explain which types its needed or valid for? If not all
> 
> Refers to: specification/network/resource-manager/Microsoft.Network/stable/2023-11-01/networkWatcher.json:4250 in [1f2768e](https://github.com/Azure/azure-rest-api-specs/commit/1f2768e3864c9dec5917626d3c8c78e58a2175a5). [](commit_id = [1f2768e](https://github.com/Azure/azure-rest-api-specs/commit/1f2768e3864c9dec5917626d3c8c78e58a2175a5), deletion_comment = False)

Added the type details in the description.

**Review by mentat9 — APPROVED** [review]
